pandas/core/base.py

```diff
@@ -1172,7 +1171,7 @@ def tolist(self):
         --------
         numpy.ndarray.tolist
         """
-        if is_datetimelike(self._values):
+        if self.dtype.kind in ["m", "M"]:
```

Comment:
I don't work super much with datetime arrays, but ``if self.dtype.kind in ["m", "M"]:`` looks very implicit, while the old way was more readable. Would it be possible to do e.g. ``needs_i8_conversion(if self.dtype)`` instead?

Reply:
the equivalent alternative would be `is_timedelta64_dtype(self._values.dtype) or is_datetime64_dtype(self._values.dtype)`.  I prefer the `dtype.kind` lookup because it is significantly more performant, but either way works

Reply:
This is not my area so much, so ok to this from me.